src/OpenLoco/Map/Tile.h

```diff
@@ -32,12 +32,69 @@ namespace OpenLoco::Map
         coord_t x = 0;
         coord_t y = 0;
 
-        map_pos() {}
-        map_pos(coord_t x, coord_t y)
+        constexpr map_pos() {}
+        constexpr map_pos(coord_t x, coord_t y)
             : x(x)
             , y(y)
         {
         }
+
+        constexpr bool operator==(const map_pos& rhs)
+        {
+            return x == rhs.x && y == rhs.y;
+        }
+
+        constexpr map_pos& operator+=(const map_pos& rhs)
+        {
+            x += rhs.x;
+            y += rhs.y;
+            return *this;
+        }
+
+        constexpr map_pos& operator-=(const map_pos& rhs)
+        {
+            x -= rhs.x;
+            y -= rhs.y;
+            return *this;
+        }
+
+        constexpr friend map_pos operator+(map_pos lhs, const map_pos& rhs)
+        {
+            lhs += rhs;
+            return lhs;
+        }
+
+        constexpr friend map_pos operator-(map_pos lhs, const map_pos& rhs)
+        {
+            lhs -= rhs;
+            return lhs;
+        }
+
+        constexpr map_pos rotate(uint8_t rotation) const
```

Comment:
Do we have some type for 90-degree rotation? Seems safer to have `Rotation` enum so we don't need to add a useless `default` case.

Reply:
do we really want to create something like that. Theres going to be lots of casts.